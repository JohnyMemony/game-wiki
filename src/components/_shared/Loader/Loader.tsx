import React from 'react';
import { Pane, Spinner } from 'evergreen-ui';

interface Props {
  centered?: boolean;
}

export const Loader: React.FC<Props> = (props) => {
  const { centered } = props;
  const marginX = centered ? 'auto' : 0;
  const marginY = centered ? 120 : 0;

  return (
    <Pane>
      <Spinner className="loader" marginX={marginX} marginY={marginY} />
    </Pane>
  );
};
