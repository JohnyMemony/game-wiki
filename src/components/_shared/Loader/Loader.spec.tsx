import React from 'react';
import { shallow } from 'enzyme';
import { Loader } from './Loader';

describe('Loader component', () => {
  it('should render Loader component', function () {
    const component = shallow(<Loader centered={false} />);
    
    expect(component).toMatchSnapshot();
  });

  it('should render centered Loader component', function () {
    const component = shallow(<Loader centered />);

    expect(component).toMatchSnapshot();
  });
});
