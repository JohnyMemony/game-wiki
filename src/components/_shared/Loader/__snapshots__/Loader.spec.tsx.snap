// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render Loader component 1`] = `
<Memo(ForwardRef(Pane))>
  <Memo(ForwardRef(Spinner))
    className="loader"
    marginX={0}
    marginY={0}
  />
</Memo(ForwardRef(Pane))>
`;

exports[`Loader component should render centered Loader component 1`] = `
<Memo(ForwardRef(Pane))>
  <Memo(ForwardRef(Spinner))
    className="loader"
    marginX="auto"
    marginY={120}
  />
</Memo(ForwardRef(Pane))>
`;
